src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,43 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    public Deadline(String description, String by, boolean isDone) {
+        super(description, isDone);
+        try {
+            this.by = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("OOPS! The deadline in an incorrect format! " +
+                    "Please indicate the date as <yyyy-mm-dd>");
+        }
+    }
+
+    public String getParsedTask() {
+        return "deadline " + this.description + " /by " + this.by + System.lineSeparator() 
+                + this.isDone + System.lineSeparator();
```

Comment:
Should the wrapped line be indented 8 spaces relative to the parent line (line 20) rather than the previous line (line 21)? (See Java coding standard -> Layout)

Reply:
Hmm, I thought that line 21 would be considered the parent line of line 22? I've seen the part of Java coding standard that you've pointed out, and I get that you're referring to the second example of:
```
if (isReady) {
    setText("Long line split"
        + "into two parts.");
}
```
However, I've also looked at parts outside of that example which seems to suggest the parent line is the line above. E.g.
```
method(param1,
        object.method()
                .method2(),
        param3);
```
I will further check with my TA regarding this and let you know the outcome! :) Thanks for pointing it out though!